### DISCUSSION TITLE
best landing pages to help QR code users subscribe to calendars?

### DISCUSSION THREAD

**Discussion by dpocock** [bitfireAT/icsx5#53] [Show and tell]
In many cases it would be useful to have a single QR code on publicity materials to help people subscribe to the calendar for some group or venue.

In practice, when people scan a QR code containing a `webcal://` link the results are unpredictable, sometimes it tries to open it like a web page or sometimes it only makes one download rather than subscribing to poll the feed.

I've created a landing page to help mitigate these problems.  You put a `http://` link in the QR code and when somebody scans it, the instructions appear.  Accessing exactly the same URL with the `webcal://` prefix returns the iCalendar data.  For Android users, it suggests installing ICSx5.  Is there anything else that could be added to the page to maximize its effectiveness?

I've shared it here so people can adapt it, convert it into other scripting languages, etc

https://gitlab.com/dpocock/calendar-subscription-page

**Comment by rfc2822** [comment]
Ah, nice.

I think when using a specific QR code scanner (like Binary Eye), the results should be sufficiently predictable :)

**Reply by dpocock** [reply]
Personally, I tested with the basic Barcode Scanner (ZXing)

When putting a QR code on a leaflet or poster it isn't always possible to get users to change the app before they actually scan the code.

**Comment by dpocock** [comment]
Owncloud / Nextcloud have features for subscribing to public webcal calendars and making them available to mobile users through CalDAV.  It would be useful to integrate this into the landing page too.
https://github.com/nextcloud/server/issues/1497
